File: js/problem002.js
function solution() {
  let sum = 0;
  let a = 1,
    b = 2;
  while (b < 4000000) {
    if (b % 2 == 0) sum += b;
    const newFib = a + b;
    a = b;
    b = newFib;
  }
  return sum;
}

console.log(solution());

File: dart/problem002.dart
int solution() {
  var a = 1, b = 2;
  var sum = 0;

  while (b < 4000000) {
    if (b.isEven) sum += b;
    var newFib = a + b;
    a = b;
    b = newFib;
  }

  return sum;
}

void main() {
  print(solution());
}
